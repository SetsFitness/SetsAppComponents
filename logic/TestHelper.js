import ably from "../redux/reducers/ablyReducer";
import auth from "../redux/reducers/authReducer";
import cache from "../redux/reducers/cacheReducer";
import info from "../redux/reducers/infoReducer";
import message from "../redux/reducers/messageReducer";
import search from "../redux/reducers/searchReducer";
import user from "../redux/reducers/userReducer";

/**
 * Class used for testing certain parts of the code that may try to connect to AWS in order to complete. Run this when
 * we are automatically running a test.
 */
export default class TestHelper {
    static ifTesting = false;
    static setTest() { TestHelper.ifTesting = true; }
    static unsetTest() { TestHelper.ifTesting = false; }
}

/**
 * Gets the entire redux store components given for initializing a mock store. Allows you to only initialize what you
 * need to for the test.
 *
 * TODO Edit when we use more reducers.
 *
 * @param {[string]} storeTypes The strings for the types of the redux store to initialize.
 * @return {{ably: *, auth: *, cache: *, info: *, message: *, user: *, search: *}}
 */
export const getInitalReduxStore = (storeTypes) => {
    return {
        ably: !storeTypes || storeTypes.includes("ably") ? ably(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        auth: !storeTypes || storeTypes.includes("auth") ? auth(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        cache: !storeTypes || storeTypes.includes("cache") ? cache(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        info: !storeTypes || storeTypes.includes("info") ? info(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        message: !storeTypes || storeTypes.includes("message") ? message(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        user: !storeTypes || storeTypes.includes("user") ? user(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        search: !storeTypes || storeTypes.includes("search") ? search(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
    }
};