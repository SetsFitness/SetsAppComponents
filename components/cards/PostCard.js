import React, {useState} from 'react';
import {Button, Card, Dimmer, Grid} from 'semantic-ui-react';
import {connect} from "react-redux";
import PostDescriptionModal from '../modals/PostDescriptionModal';
import {Player} from "video-react";
import ItemType from "../../logic/ItemType";
import ChallengeDetailCard, {ChallengeDetailCardInfo} from "./post_detail_cards/ChallengeDetailCard";
import PostDetailCard from "./post_detail_cards/PostDetailCard";
import ClientDetailCard from "./post_detail_cards/ClientDetailCard";
import TrainerDetailCard from "./post_detail_cards/TrainerDetailCard";
import {convertFromISO} from "../../logic/TimeHelper";
import ClientModal from "../modals/ClientModal";
import TrainerModal from "../modals/TrainerModal";
import {GroupDetailCardInfo} from "./post_detail_cards/GroupDetailCard";
import {EventDetailCardInfo} from "./post_detail_cards/EventDetailCard";
import Spinner from "../props/Spinner";
import {getAttributeFromObject, getObjectAttribute, getObjectFromCache} from "../../logic/CacheRetrievalHelper";
import StyledProfileImage from "../props/StyledProfileImage";
import type Post from "../../types/Post";

export const PostCardInfo = {
  fetchList: ["id", "time_created", "by", "item_type", "postType", "about", "access", "description", "videoPaths", "picturePaths"],
  byInfo: {
    clientFetchList: ["id", "name", "friends", "challengesWon", "scheduledEvents", "profileImagePath"],
    trainerFetchList: ["id", "name", "gender", "birthday", "profileImagePath", "profileImagePaths"],
  },
  aboutInfo: {
    clientFetchList: ["id", "name", "friends", "challengesWon", "scheduledEvents", "profileImagePath"],
    trainerFetchList: ["id", "name", "gender", "birthday", "profileImagePath", "profileImagePaths"],
    eventFetchList: EventDetailCardInfo.fetchList,
    challengeFetchList: ChallengeDetailCardInfo.fetchList,
    postFetchList: ["id", "time_created", "by", "item_type", "postType", "about", "access", "description", "videoPaths", "picturePaths"],
    groupFetchList: GroupDetailCardInfo.fetchList,
  },
  ifSubscribe: false,
};

type Props = {
  post: Post
  // by: {
  //     id: string,
  //     item_type: string,
  //     name: string,
  //     profileImagePath: string,
  //     profileImage: string,
  // },
  // about?: {
  //     id: string,
  //     item_type: string,
  //     title: string,
  //     owner: string
  // }
};

/**
 * Displays the profile image.
 *
 * @param {string} profileImage The url string for the profile image.
 * @returns {*} Displays the JSX of the profile image given.
 */
export const profilePicture = (profileImage) => {
  if (profileImage) {
    return (
      <StyledProfileImage profileImage={profileImage} type="Small"/>
      //<div align="center" className="ui u-avatar tiny" style={{backgroundImage: `url(${profileImage})`, width: '50px', height: '50px'}}/>
    );
  } else {
    return (
      <Dimmer inverted>
        <Spinner className="spinner1"/>
      </Dimmer>
    );
  }
};

/**
 * Displays the video or image media in the post.
 *
 * @param {[]} pictures A list of all the pictures in the post.
 * @param {[]} videos A list of all the videos in the post.
 * @returns {*} The React JSX used to display the video or image in a post.
 */
export const getPostMedia = (pictures, videos) => {
  // TODO FIX THIS UP SO THAT IT CAN HANDLE A LOT MORE POSTS
  if (videos && videos.length > 0) {
    return (
      <Player inline={true}>
        <source src={videos[0]} type="video/mp4"/>
      </Player>
    );
  }
  return null;
};

/**
 * Displays the correct detail card based on the type of post that it is.
 *
 * @param {string} postType The type of post that is being displayed.
 * @param {string} about The id of the content that the post is about.
 * @returns {*}
 */
export const getCorrectDetailCard = (postType, about) => {
  if (postType && postType.length) {
    //alert("Item Type: " + itemType);
    if (postType.substr(0, 3) === "new") {
      // TODO This indicates that this is for a newly created Item
      postType = ItemType[postType.substring(3, postType.length)];
    }
    //alert(itemType);
    if (postType) {
      //alert("Post Type: " + postType);
      // TODO Switch the post types
      if (postType === "Client") {
        //alert("Client Share Post Spotted!");
        // if(!this.state.postMessageSet) {
        //     this.setState({postMessage: "shared a user profile", postMessageSet: true});
        // }
        return (<ClientDetailCard client={about}/>);
      } else if (postType === "Trainer") {
        // if(!this.state.postMessageSet) {
        //     this.setState({postMessage: "shared a trainer profile", postMessageSet: true});
        // }
        return (<TrainerDetailCard trainer={about}/>);
      } else if (postType === "Gym") {
        //return (<GymDetailCard displayMedia = {this.getDisplayMedia}/>);
      } else if (postType === "Workout") {
        //return (<WorkoutDetailCard displayMedia = {this.getDisplayMedia}/>);
      } else if (postType === "Review") {
        //return (<ReviewDetailCard displayMedia = {this.getDisplayMedia}/>);
      } else if (postType === "Event") {
        //return (<EventDetailCard displayMedia = {this.getDisplayMedia}/>);
      } else if (postType === "Challenge") {
        // if(!this.state.postMessageSet) {
        //     this.setState({postMessage: "shared a challenge", postMessageSet: true});
        // }
        return (<ChallengeDetailCard challenge={about}/>);
      } else if (postType === "Invite") {
        //return (<InviteDetailCard displayMedia = {this.getDisplayMedia}/>);
      } else if (postType === "Post") {
        // if(!this.state.postMessageSet) {
        //     this.setState({postMessage: "posted", postMessageSet: true});
        // }
        return (<PostDetailCard postID={about.id}/>);
      }
    }
  }
  return (<div/>);
};

/**
 * The modal of the user that the post is authored by.
 *
 * @param {string} id The id of the user that created the modal.
 * @param {string} itemType
 * @param {function()} byModalOpen Opens the modal relating to the user who authored the post.
 * @param {function(boolean)} setByModalOpen Sets the by modal open attribute to true.
 * @returns {*} The React JSX user to display the proper client modal state.
 */
export const byModal = (id, itemType, byModalOpen, setByModalOpen) => {
  if (itemType === "Client") {
    return (
      <ClientModal open={byModalOpen} onClose={() => setByModalOpen(false)} clientID={id}/>
    );
  } else if (itemType === "Trainer") {
    return (
      <TrainerModal open={byModalOpen} onClose={() => setByModalOpen(false)} trainerID={id}/>
    );
  }
  return null;
};

/**
 * This is the generic view for how a post shows up in any feeds or lists.
 * It is used as a modal trigger in the feed.
 *
 * @param {Props} props The given props to the component.
 * @returns {*} The React JSX used to display the component.
 * @constructor
 */
const PostCard = (props: Props) => {
  const [postModalOpen, setPostModalOpen] = useState(false);
  const [byModalOpen, setByModalOpen] = useState(false);

  const getPostAttribute = (attributeName) => {
    return getAttributeFromObject(props.post, attributeName);
  };
  const getByAttribute = (attributeName) => {
    return getObjectAttribute(getPostAttribute("by"), attributeName, props.cache);
  };
  // const getAboutAttribute = (attributeName) => {
  //     return getAttributeFromObject(props.about, attributeName, props.cache);
  // };
  // const getBy = () => {
  //     return getObjectFromCache(props.by, props.cache);
  // };
  const getAbout = () => {
    return getObjectFromCache(getPostAttribute("about"), props.cache);
  };

  if (!props.post || !getByAttribute("name")) {
    return (
      <Card color='purple' fluid raised>
        <Spinner loading={true}/>
      </Card>
    );
  } else {
    return (
      <Card color='purple' fluid raised>
        {/*this.getPostAttribute("about")*/}
        {/*{openOnce()}*/}
        <Card.Content>
          <div align='center'>
            {getCorrectDetailCard(getPostAttribute("postType"), getAbout())}
          </div>
          {/*this.getDisplayMedia()*/}
        </Card.Content>
        <Card.Content extra onClick={() => postModalOpen || setPostModalOpen(true)}>
          {/*<Card.Meta textAlign = 'center'>{this.getPostAttribute("description")}</Card.Meta>*/}
          <div align="center">
            {convertFromISO(getPostAttribute("time_created")).substr(5, 12)}
          </div>
          <PostDescriptionModal open={postModalOpen} onClose={() => setPostModalOpen(false)}
                                postID={props.post.id}/>
        </Card.Content>
      </Card>
    );
  }
};

const mapStateToProps = state => ({
  cache: state.cache
});

// const connector: Connector<Props, any> = connect(mapStateToProps);
// export default connector(PostCard);
export default connect(mapStateToProps)(PostCard);
