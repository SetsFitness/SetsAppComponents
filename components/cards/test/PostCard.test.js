import React from 'react';
import { Provider } from 'react-redux';
import "../../../testing/SetTesting";
import { expect } from 'chai';
import { store, funCompWrapper } from "../../../testing/TestHelper";
import { shallow, mount, render } from 'enzyme';
import PostCard, {byModal, getCorrectDetailCard, profilePicture} from "../PostCard";
import ClientDetailCard from "../post_detail_cards/ClientDetailCard";
import PostDetailCard from "../post_detail_cards/PostDetailCard";
// import ClientModal from "../modals/ClientModal";
// import TrainerModal from "../modals/TrainerModal";

const reduxState = {
    cache: {
        clients: {
            CL0001: {
                id: "CL0001",
                profileImage: "thisIsAFakePath/Path",
                name: "Sleeny"
            }
        },
        challenges: {
            CH0001: {
                id: "CH0001",
                by: "CL0001",
            }
        }
    },
};

const post = {
    id: "PO0001",
    by: "CL0001",
    about: "CH0001",
    postType: "Client",
    time_created: "2016-09-15T15:53:00"

};
const post1 = {
    id: "PO0001",
    by: "CL0001",
    about: "CH0001",
    postType: "Trainer"
};
const post2 = {
    id: "PO0001",
    by: "CL0001",
    about: "Challenge"
};
const post3 = {
    id: "PO0001",
    by: "CL0001",
    about: "Post"
};

const invalidPost = {
    id: "PO0001",
    about: "invalid"
};

// ====================================================================================================================
// =                                        COMPONENT LEVEL TESTS                                                     =
// ====================================================================================================================

describe("Post Card", () => {
    // Props ~
    describe("Props", () => {
        // Pass in the props into the component and make sure that they show up in the component's props.
        // Also make sure if you don't pass in the props directly that every prop has a default prop.
        // (We haven't been paying a whole lot of attention to the default props, so we'll have to edit the components
        // to fix that...)
    });
    // States ~
    describe("States", () => {
        // Here you check the state of the component, like the shown text and sub-components for the component.
        // Make sure you go into all the conditions of the component so that you test every possible state of the
        // component.
        // EXAMPLE: it("Shows the spinner when no post is passed in");
    });
    // Events ~
    describe("Events", () => {
        // Here you use .simultate('click') to check the change in the component.
        // Only in special circumstances should you use jest.fn() and that's if you're directly passing in a function
        // into the component's props and you have access to it.
    });
});

// ====================================================================================================================
// =                                        INTERNAL FUNCTION TESTS                                                   =
// ====================================================================================================================

describe("Profile Picture", () => {
    describe("Props", () => {
        // Pass in the props into the component and make sure that they show up in the component's props.
        // Also make sure if you don't pass in the props directly that every prop has a default prop.
        // (We haven't been paying a whole lot of attention to the default props, so we'll have to edit the components
        // to fix that...)
    });
    // States ~
    describe("States", () => {
        // Here you check the state of the component, like the shown text and sub-components for the component.
        // Make sure you go into all the conditions of the component so that you test every possible state of the
        // component.
        // EXAMPLE: it("Shows the spinner when no post is passed in");
    });
    // Events ~
    describe("Events", () => {
        // Here you use .simultate('click') to check the change in the component.
        // Only in special circumstances should you use jest.fn() and that's if you're directly passing in a function
        // into the component's props and you have access to it.
    });
});
describe("By Modal", () => {
    describe("Props", () => {
        // Pass in the props into the component and make sure that they show up in the component's props.
        // Also make sure if you don't pass in the props directly that every prop has a default prop.
        // (We haven't been paying a whole lot of attention to the default props, so we'll have to edit the components
        // to fix that...)
    });
    // States ~
    describe("States", () => {
        // Here you check the state of the component, like the shown text and sub-components for the component.
        // Make sure you go into all the conditions of the component so that you test every possible state of the
        // component.
        // EXAMPLE: it("Shows the spinner when no post is passed in");
    });
    // Events ~
    describe("Events", () => {
        // Here you use .simultate('click') to check the change in the component.
        // Only in special circumstances should you use jest.fn() and that's if you're directly passing in a function
        // into the component's props and you have access to it.
    });
});
describe("Get Correct Detail Card", () => {
    describe("Props", () => {
        // Pass in the props into the component and make sure that they show up in the component's props.
        // Also make sure if you don't pass in the props directly that every prop has a default prop.
        // (We haven't been paying a whole lot of attention to the default props, so we'll have to edit the components
        // to fix that...)
    });
    // States ~
    describe("States", () => {
        // Here you check the state of the component, like the shown text and sub-components for the component.
        // Make sure you go into all the conditions of the component so that you test every possible state of the
        // component.
        // EXAMPLE: it("Shows the spinner when no post is passed in");
    });
    // Events ~
    describe("Events", () => {
        // Here you use .simultate('click') to check the change in the component.
        // Only in special circumstances should you use jest.fn() and that's if you're directly passing in a function
        // into the component's props and you have access to it.
    });
});

it("renders without crashing", () => {
    const component = shallow(<PostCard post={post} store={store(reduxState)}/>);

    global.expect(component).toMatchSnapshot();
});

// Old stuff to refactor

describe("Profile Picture", () => {
    it("profilePicture function displays spinner when there is no profile picture", () => {
        const Wrapper = funCompWrapper(profilePicture(null));

        const component = shallow(<Wrapper/>);

        expect(component.find("Spinner")).to.have.lengthOf(1);
    });

    it("profilePicture function displays image when there is a profile picture", () => {
        const Wrapper = funCompWrapper(profilePicture("test/path"));

        const component = shallow(<Wrapper/>);

        expect(component.find("StyledProfileImage")).to.have.lengthOf(1);
    });
});

describe('getCorrectDetailCard', () => {
    it("getCorrectDetailCard displays a client detail card with postType Client.", () => {
        const Wrapper = funCompWrapper(getCorrectDetailCard("Client", "CL001"));

        const component = shallow(<Wrapper/>);

        expect(component.find(ClientDetailCard)).to.have.lengthOf(1);
    });
    it("getCorrectDetailCard displays a trainer detail card with postType Trainer.", () => {
        const Wrapper = funCompWrapper(getCorrectDetailCard("Trainer", "TR001"));

        const component = shallow(<Wrapper/>);

        expect(component.find("TrainerDetailCard")).to.have.lengthOf(1);
    });
    it("getCorrectDetailCard displays a challenge detail card with postType Challenge.", () => {
        const Wrapper = funCompWrapper(getCorrectDetailCard("Challenge", "CL001"));

        const component = shallow(<Wrapper/>);

        expect(component.find("ChallengeDetailCard")).to.have.lengthOf(1);
    });
    it("getCorrectDetailCard displays a post detail card with postType Post.", () => {
        const Wrapper = () => getCorrectDetailCard("Post", "PO001");

        const component = shallow(<Wrapper/>);

        expect(component.find("Connect(PostDetailCard)")).to.have.lengthOf(1);
    });
    it("getCorrectDetailCard displays a div with an invalid postType.", () => {
        const component = shallow(getCorrectDetailCard());

        expect(component.find("div")).to.have.lengthOf(1);
    });
});

describe("byModal", () => {
    it("getCorrectDetailCard displays a challenge detail card with postType Challenge.", () => {
        const Wrapper = funCompWrapper(byModal("CL001", "Client", true, () => {}));

        const component = shallow(<Wrapper/>);

        expect(component.find("Connect(ClientModal)")).to.have.lengthOf(1);
    });

    it("getCorrectDetailCard displays a post detail card with postType Post.", () => {
        const Wrapper = funCompWrapper(byModal("TR001", "Trainer", true, () => {}));

        const component = shallow(<Wrapper/>);

        expect(component.find("Connect(TrainerModal)")).to.have.lengthOf(1);
    });
});

describe("PostCard", () => {

    describe("Props", () => {
        // Pass in the props into the component and make sure that they show up in the component's props.
        // Also make sure if you don't pass in the props directly that every prop has a default prop.
        // (We haven't been paying a whole lot of attention to the default props, so we'll have to edit the components
        // to fix that...)

        it("Post Card displays PostDescriotionModal if there is a valid post", () => {
            const component = shallow(<PostCard post={post} store={store(reduxState)}/>).shallow();

            expect(component.find("PostDescriptionModal")).to.have.lengthOf(1);
        });

        it("Post Card displays spinner if there is no name attribute in the post owner", () => {
            const component = shallow(<PostCard post={post} store={store(reduxState)}/>).shallow();

            expect(component.find("byModal")).to.have.lengthOf(1);
        });

        it("Post Card displays spinner if there is no post attribute", () => {
            const component = shallow(<PostCard post={null} store={store(reduxState)}/>).shallow();

            expect(component.find("Spinner")).to.have.lengthOf(1);
        });
    });

    // States ~
    describe("States", () => {
        // Here you check the state of the component, like the shown text and sub-components for the component.
        // Make sure you go into all the conditions of the component so that you test every possible state of the
        // component.
        // EXAMPLE: it("Shows the spinner when no post is passed in");

        it("Post Card displays spinner if there is no name attribute in the post owner", () => {
            const component = shallow(<PostCard post={invalidPost} store={store(reduxState)}/>).shallow();

            expect(component.find("Spinner")).to.have.lengthOf(1);
        });

        it("The name of the owner is displayed in the post card", () => {
            const component = shallow(<PostCard post={post} store={store(reduxState)}/>).shallow();

            const name = component.find("GridColumn").at(1).render().text();

            console.log(name);

            expect(name).equal("Sleeny  ");
        });

        it("The date that the post was created is displayed in the post card", () => {
            const component = shallow(<PostCard post={post} store={store(reduxState)}/>).shallow();

            const date = component.find("div").at(1).render().text();

            console.log(date);

            expect(date).equal("Sep 15, 2016");
        });

        it("The access of the post is displayed in the post card", () => {
            const component = shallow(<PostCard post={post} store={store(reduxState)}/>).shallow();

            const date = component.find("CardMeta").at(1).render().text();

            console.log(date);

            expect(date).equal("Public");
        });
    });

    // Events ~
    describe("Events", () => {
        // Here you use .simultate('click') to check the change in the component.
        // Only in special circumstances should you use jest.fn() and that's if you're directly passing in a function
        // into the component's props and you have access to it.

        //TODO: Figure out the best way to simulate button clicks
        it("button click should open by modal", () => {
            const component = shallow(<PostCard post={post} store={store(reduxState)}/>).shallow();
            component
                .find("Button")
                .simulate("click");

            expect(component.shallow().find("byModal")).to.have.lengthOf(1);
        });
    });

});