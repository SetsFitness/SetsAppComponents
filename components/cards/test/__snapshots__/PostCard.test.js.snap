// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard Button renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(PostCard)
    post={
      Object {
        "about": "CH0001",
        "by": "CL0001",
        "id": "PO0001",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "cache": Object {
        "challenges": Object {
          "CH0001": Object {
            "by": "CL0001",
            "id": "CH001",
          },
        },
        "clients": Object {
          "CL0001": Object {
            "id": "CL0001",
          },
        },
      },
      "dispatch": [Function],
      "post": Object {
        "about": "CH0001",
        "by": "CL0001",
        "id": "PO0001",
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "cache": Object {
          "challenges": Object {
            "CH0001": Object {
              "by": "CL0001",
              "id": "CH001",
            },
          },
          "clients": Object {
            "CL0001": Object {
              "id": "CL0001",
            },
          },
        },
        "dispatch": [Function],
        "post": Object {
          "about": "CH0001",
          "by": "CL0001",
          "id": "PO0001",
        },
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
